src/test/13_ProxyUpgrade.t.sol

```diff
@@ -0,0 +1,816 @@
+                oracleAddresses: oracleInterleaveToEthEthToUsd,
+                id: 1,
+                assetAddress: address(interleave)
+            }),
+            assetCreditRatings
+        );
+
+        liquidator = new Liquidator(
+            0x0000000000000000000000000000000000000000,
+            address(mainRegistry)
+        );
+        vm.stopPrank();
+
+        vm.startPrank(vaultOwner);
+        vault = new Vault();
+        stable.transfer(address(0), stable.balanceOf(vaultOwner));
+        vm.stopPrank();
+
+        vm.startPrank(creatorAddress);
+        factory = new Factory();
+        factory.setNewVaultInfo(
+            address(mainRegistry),
+            address(vault),
+            stakeContract,
+            address(interestRateModule),
+            Constants.upgradeProof1To2
+        );
+        factory.confirmNewVaultInfo();
+        factory.setLiquidator(address(liquidator));
+        liquidator.setFactory(address(factory));
+        mainRegistry.setFactory(address(factory));
+        mainRegistry.setFactory(address(factory));
+        vm.stopPrank();
+
+        vm.startPrank(tokenCreatorAddress);
+        stable.setLiquidator(address(liquidator));
+        stable.setFactory(address(factory));
+        vm.stopPrank();
+
+        vm.prank(vaultOwner);
+        proxyAddr = factory.createVault(
+            uint256(
+                keccak256(
+                    abi.encodeWithSignature(
+                        "doRandom(uint256,uint256,bytes32)",
+                        block.timestamp,
+                        block.number,
+                        blockhash(block.number)
+                    )
+                )
+            ),
+            Constants.UsdNumeraire
+        );
+        proxy = Vault(proxyAddr);
+
+        vm.prank(address(proxy));
+        stable.mint(tokenCreatorAddress, 100000 * 10**Constants.stableDecimals);
+
+        vm.startPrank(oracleOwner);
+        oracleEthToUsd.transmit(int256(rateEthToUsd));
+        oracleLinkToUsd.transmit(int256(rateLinkToUsd));
+        oracleSnxToEth.transmit(int256(rateSnxToEth));
+        oracleWbaycToEth.transmit(int256(rateWbaycToEth));
+        oracleWmaycToUsd.transmit(int256(rateWmaycToUsd));
+        oracleInterleaveToEth.transmit(int256(rateInterleaveToEth));
+        vm.stopPrank();
+
+        vm.startPrank(vaultOwner);
+        bayc.setApprovalForAll(address(proxy), true);
+        mayc.setApprovalForAll(address(proxy), true);
+        dickButs.setApprovalForAll(address(proxy), true);
+        interleave.setApprovalForAll(address(proxy), true);
+        eth.approve(address(proxy), type(uint256).max);
+        link.approve(address(proxy), type(uint256).max);
+        snx.approve(address(proxy), type(uint256).max);
+        safemoon.approve(address(proxy), type(uint256).max);
+        stable.approve(address(proxy), type(uint256).max);
+        stable.approve(address(liquidator), type(uint256).max);
+        vm.stopPrank();
+    }
+
+
+    /* 
```

Comment:
Remove all your garbage before PRs plz

Reply:
must stay there for readability ser, otherwise no-one knows where the proofs come from

Reply:
In Constants?